### PR TITLE
classifier drift

### PR DESCRIPTION
**Pull request by arnaudvl**
To do:

- [x] Classifier drift method
- [x] Example notebook
- [x] Save/load fn
- [x] Tests
- [x] Docs

### CODE REVIEW

File: examples/cd_clf_cifar10.ipynb [hunk 1]
```diff
@@ -0,0 +1,556 @@
+{
```
Comment (jklaise) [inline examples/cd_clf_cifar10.ipynb:1]: <p>I would probably import the <code>f1_score</code>  here as then it's in context, otherwise have to scroll to the top to check where it comes from.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi-detect/pull/169/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

**Review by jklaise — CHANGES_REQUESTED** [review]

File: alibi_detect/cd/classifier.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            0 is silent, 1 a progress bar and 2 prints the statistics after each epoch.
+        fit_kwargs
+            Optional additional kwargs when fitting the classifier.
+        data_type
+            Optionally specify the data type (tabular, image or time-series). Added to metadata.
+        """
+        super().__init__()
+
+        if threshold is None:
+            logger.warning('Need to set drift threshold to detect data drift.')
```
[the remark this answers is not in the document]
Comment (arnaudvl) [reply alibi_detect/cd/classifier.py:96]: This is the same as for other drift, outlier and adversarial detectors and allows for easier saving/loading. Part of the revamp of all the save/load functionality would be to take this into account.

File: alibi_detect/cd/classifier.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        self.X_ref = update_reference(self.X_ref, X, self.n, self.update_X_ref)
+        # used for reservoir sampling
+        self.n += X.shape[0]  # type: ignore
+
+        # populate drift dict
+        cd = concept_drift_dict()
+        cd['meta'] = self.meta
+        cd['data']['is_drift'] = drift_pred
+        if return_metric:
+            cd['data'][f'{self.metric_fn.__name__}'] = drift_metric
```
[the remark this answers is not in the document]
Comment (arnaudvl) [reply alibi_detect/cd/classifier.py:235]: Was also a bit unsure about this, had a default field `metric` initially. But then thought it is actually more informative to actually return the metric name that is being computed. Maybe downstream applications should know what exactly they are using?

[the remark this answers is not in the document]
Comment (jklaise) [reply alibi_detect/cd/classifier.py:235]: Hmm, maybe we can return 2 things, `metric` and `metric_name` ?

File: alibi_detect/cd/classifier.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        else:  # use stratified folds
+            splits = self.skf.split(x, y)
+
+        # iterate over folds: train a new model for each fold and make out-of-fold (oof) predictions
+        preds_oof, idx_oof = [], []
+        for idx_tr, idx_te in splits:
+            x_tr, y_tr, x_te = x[idx_tr], np.eye(2)[y[idx_tr]], x[idx_te]
+            clf = tf.keras.models.clone_model(self.model)
+            clf.compile(**self.compile_kwargs)
+            self.fit_kwargs.update({'x': x_tr, 'y': y_tr})
```
[the remark this answers is not in the document]
Comment (arnaudvl) [reply alibi_detect/cd/classifier.py:191]: Good point, will adjust this.

**Review by jklaise — APPROVED** [review]
